diff --git a/node_modules/react-native-pager-view/ios/ReactNativePageView.m b/node_modules/react-native-pager-view/ios/ReactNativePageView.m
index 78f266b..95671d8 100644
--- a/node_modules/react-native-pager-view/ios/ReactNativePageView.m
+++ b/node_modules/react-native-pager-view/ios/ReactNativePageView.m
@@ -75,6 +75,8 @@ - (void)didMoveToSuperview {
         [self embed];
         [self setupInitialController];
     }
+    
+  
 }
 
 - (void)didMoveToWindow {
@@ -83,6 +85,11 @@ - (void)didMoveToWindow {
         [self embed];
         [self setupInitialController];
     }
+    
+    if (self.reactViewController.navigationController != nil && self.reactViewController.navigationController.interactivePopGestureRecognizer != nil) {
+
+  [self.scrollView.panGestureRecognizer requireGestureRecognizerToFail:self.reactViewController.navigationController.interactivePopGestureRecognizer];
+  }
 }
 
 - (void)embed {
@@ -161,6 +168,8 @@ - (void)setupInitialController {
                             direction:UIPageViewControllerNavigationDirectionForward
                              animated:YES
              shouldCallOnPageSelected:YES];
+              
+         
     }
 }
 
